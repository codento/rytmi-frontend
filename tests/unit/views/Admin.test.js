import Vuex from 'vuex'
import BootstrapVue from 'bootstrap-vue'
import { mount, createLocalVue } from '@vue/test-utils'
import View from '@/views/Admin.vue'

describe('Admin.vue', () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  localVue.use(BootstrapVue)

  let store, getters, wrapper

  beforeEach(() => {
    getters = {
      users: () => {
        return {
          1: {
            id: 1,
            firstName: 'Bar',
            lastName: 'Foo',
            active: true,
            admin: true
          },
          2: {
            id: 2,
            firstName: 'Oof',
            lastName: 'Rab',
            active: true,
            admin: true
          },
          3: {
            id: 3,
            firstName: 'Abr',
            lastName: 'Ofo',
            active: false,
            admin: true
          },
          4: {
            id: 4,
            firstName: 'Rba',
            lastName: 'Off',
            active: true,
            admin: false
          }
        }
      },
      profileByUserId: () => (arg) => {
        return {
          id: arg,
          userId: 1,
          firstName: 'Bar',
          lastName: 'Foo',
          photoPath: '',
          employeeRoles: [1],
          title: 'software developer',
          email: 'foo.bar@foo.com',
          phone: '1354',
          description: 'fdas'
        }
      },
      employeeRoles: () => {
        return [
          { id: 1, title: 'soft mengineer' },
          { id: 2, title: 'hardcore soft engineer' }
        ]
      },
      absencesByProfileId: () => (id) => {},
      leaves: () => []
    }
    const actions = {
      fetchAbsencesForProfile: () => (id) => {}
    }

    store = new Vuex.Store({ getters, actions })
    wrapper = mount(View, {
      store,
      localVue
    })
  })

  it('Template is correct', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('Renders user list with correct amount of users', () => {
    const users = wrapper.findAll('.list-group-item')
    expect(users.length).toBe(4)
  })

  it('Renders renders EditUser when a user is selected', () => {
    wrapper.vm.setUser(1)
    expect(wrapper.find('#edit-user-card').exists()).toBeTruthy()
  })
})
