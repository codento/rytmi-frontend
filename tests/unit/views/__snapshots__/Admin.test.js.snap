// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin.vue Template is correct 1`] = `
<div
  class="container"
>
  <h2>
    Admin
  </h2>
   
  <div
    class="row"
  >
    <div
      class="col-sm-12 col-md-6"
    >
      <div>
        <fieldset
          class="form-group"
          id="__BVID__7"
        >
          <legend
            class="col-form-label pt-0"
            id="__BVID__7__BV_label_"
            tabindex="-1"
          >
            Filter by name
          </legend>
          <div
            role="group"
            tabindex="-1"
          >
            <input
              class="form-control"
              id="__BVID__8"
              type="text"
            />
            <!---->
            <!---->
            <!---->
          </div>
        </fieldset>
         
        <div>
          
    Active User
          <span
            class="cui-check"
          />
          
    Admin User
          <span
            class="cui-cog"
          />
        </div>
         
        <div
          class="list-group"
        >
          <div
            class="list-group-item"
            name="select-user"
          >
            
      Abr Ofo
      
            <!---->
             
            <span
              class="cui-cog"
            />
          </div>
        </div>
        <div
          class="list-group"
        >
          <div
            class="list-group-item"
            name="select-user"
          >
            
      Bar Foo
      
            <span
              class="cui-check"
            />
             
            <span
              class="cui-cog"
            />
          </div>
        </div>
        <div
          class="list-group"
        >
          <div
            class="list-group-item"
            name="select-user"
          >
            
      Oof Rab
      
            <span
              class="cui-check"
            />
             
            <span
              class="cui-cog"
            />
          </div>
        </div>
        <div
          class="list-group"
        >
          <div
            class="list-group-item"
            name="select-user"
          >
            
      Rba Off
      
            <span
              class="cui-check"
            />
             
            <!---->
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="col-sm-12 col-md-6"
    >
      <div
        id="edit-user-card"
      >
        <!---->
         
        <!---->
      </div>
    </div>
  </div>
</div>
`;
